peps/pep-0728.rst

```diff
@@ -0,0 +1,633 @@
+
+The following examples illustrate these checks in action.
+
+``__extra__`` as a pseudo-field follows the same rules that other fields have,
+so when both TypedDict contains ``__extra__``, this check is naturally
+enforced::
+
+    class MovieExtraInt(TypedDict):
+        name: str
+        __extra__: int
+
+    class MovieExtraStr(TypedDict):
+        name: str
+        __extra__: str
+    
+    extra_int: MovieExtraInt = {"name": "No Country for Old Men", "year": 2007}
+    extra_str: MovieExtraStr = {"name": "No Country for Old Men", "description": ""}
+    extra_int = extra_str  # Not OK. 'str' is inconsistent with 'int' for field '__extra__'
+    extra_str = extra_int  # Not OK. 'int' is inconsistent with 'str' for field '__extra__'
+    
+When only one of the TypedDict ``t`` has ``__extra__``, then the keys unknown to
+``t`` in the other TypedDict corresponds to the ``__extra__`` pseudo-field.
+Still, the same set of rules from :pep:`589` apply::
+
+    class MovieNotRequiredYear(TypedDict):
+        name: str
+        year: NotRequired[int]
+    
+    class MovieRequiredYear(TypedDict):
+        name: str
+        year: int
+
+    extra_int: MovieExtraInt = {"name": "Kill Bill Vol. 1"}
+    required_year: MovieNotRequiredYear = {"name": "Kill Bill Vol. 1"}
+    not_required_year: MovieRequiredYear = {"name": "Kill Bill Vol. 1", "year": 2003}
+
+    extra_int = required_year  # Not OK. Having an extra required field makes
+                               # 'required_year' structurally incompatible with
+                               # 'extra_int'
+    extra_int = not_required_year  # OK.
+
+    required_year = extra_int  # Not OK. The required key 'year' is missing in
+                               # 'extra_int'
+    not_required_year = extra_int  # OK.
+
+A TypedDict can be consistent with ``Mapping[...]`` types other than
+``Mapping[str, object]`` as long as the union of value types on all fields is
+consistent with the value type of the ``Mapping[...]`` type. It is an extension
+to this rule::
+
+    * A TypedDict with all ``int`` values is not consistent with
+      ``Mapping[str, int]``, since there may be additional non-``int``
+      values not visible through the type, due to structural subtyping.
+      These can be accessed using the ``values()`` and ``items()``
+      methods in ``Mapping``
+
+For example::
+
+    extra_str: MovieExtraStr = {"name": "Blade Runner", "summary": ""}
+    str_mapping: Mapping[str, str] = extra_str  # OK
+
+    int_mapping: Mapping[str, int] = extra_int  # Not OK. 'int | str' is not consistent with 'int'
+    int_str_mapping: Mapping[str, int | str] = extra_int  # OK
+
+Note that because the presence of ``__extra__`` prohibits required keys from
+being added through inheritance, it is possible to determine if a TypedDict type
+and its structural subtypes will never have any required key during static
+analysis.
+
+If there is no required key, the TypedDict type is consistent with ``Dict[...]``
```

Comment:
`Dict` is deprecated, use `dict`. Maybe write `dict[KT, VT]` so you have a way to talk about `VT` instead of "the value type".